### PR TITLE
add measure to Cylinder and CylinderSurface

### PR DESCRIPTION
**Pull request by dorn-gerhard**

### CODE REVIEW

**Review by juliohm — CHANGES_REQUESTED** [review]
Awesome. We are only missing tests.

**Comment by dorn-gerhard** [comment]
I added tests for measure of Cylinder and CylinderSurface

File: src/primitives/cylinder.jl [hunk 1]
```diff
@@ -56,4 +56,6 @@ axis(c::Cylinder) = axis(boundary(c))
 
 isright(c::Cylinder) = isright(boundary(c))
 
-boundary(c::Cylinder) = CylinderSurface(c.radius, c.bot, c.top)
\ No newline at end of file
+boundary(c::Cylinder) = CylinderSurface(c.radius, c.bot, c.top)
+
+measure(c::Cylinder) = norm(origin(c.bot) - origin(c.top)) * c.radius^2 * pi
```
Comment (juliohm) [inline src/primitives/cylinder.jl]:
```suggestion
measure(c::Cylinder) = norm(origin(c.bot) - origin(c.top)) * c.radius^2 * pi

volume(c::Cylinder) = measure(c)
```

File: src/primitives/cylindersurface.jl [hunk 2]
```diff
@@ -70,4 +70,6 @@ function isright(c::CylinderSurface{T}) where {T}
   isparallelv && isparallelw
 end
 
-boundary(::CylinderSurface) = nothing
\ No newline at end of file
+boundary(::CylinderSurface) = nothing
+
+measure(c::CylinderSurface) = (norm(origin(c.bot) - origin(c.top)) + c.radius) * 2 * c.radius * pi
```
Comment (juliohm) [inline src/primitives/cylindersurface.jl]:
```suggestion
measure(c::CylinderSurface) = (norm(origin(c.bot) - origin(c.top)) + c.radius) * 2 * c.radius * pi

area(c::CylinderSurface) = measure(c)
```

File: test/primitives.jl [hunk 3]
```diff
@@ -285,6 +285,7 @@
     @test axis(c) == Line(P3(1,2,3), P3(4,5,6))
     @test isconvex(c)
     @test !isright(c)
+    @test measure(c) ≈ T(5)^2 * pi * T(3)*sqrt(T(3))
```
Comment (juliohm) [inline test/primitives.jl]:
```suggestion
    @test measure(c) == volume(c) ≈ T(5)^2 * pi * T(3)*sqrt(T(3))
```

File: test/primitives.jl [hunk 4]
```diff
@@ -302,6 +303,7 @@
     @test axis(c) == Line(P3(0,0,0), P3(0,0,1))
     @test isright(c)
     @test boundary(c) == CylinderSurface(T(1), Segment(P3(0,0,0), P3(0,0,1)))
+    @test measure(c) ≈ pi
```
Comment (juliohm) [inline test/primitives.jl]:
```suggestion
    @test measure(c) == volume(c) ≈ pi 
```

File: test/primitives.jl [hunk 5]
```diff
@@ -318,6 +320,12 @@
     @test isconvex(c)
     @test isright(c)
     @test isnothing(boundary(c))
+    @test measure(c) ≈ 2 * T(2)^2 * pi + 2 * T(2) * pi 
+
+    c = CylinderSurface(T(5),
+                 Plane(P3(1,2,3), V3(0,0,1)),
+                 Plane(P3(4,5,6), V3(0,0,1)))
+    @test measure(c) ≈ 2 * T(5)^2 * pi + 2 * T(5) * pi * sqrt(3*T(3)^2)
```
Comment (juliohm) [inline test/primitives.jl]:
```suggestion
    @test measure(c) == area(c) ≈ 2 * T(2)^2 * pi + 2 * T(2) * pi 

    c = CylinderSurface(T(5),
                 Plane(P3(1,2,3), V3(0,0,1)),
                 Plane(P3(4,5,6), V3(0,0,1)))
    @test measure(c) == area(c) ≈ 2 * T(5)^2 * pi + 2 * T(5) * pi * sqrt(3*T(3)^2)
```

**Pull request merged**